usvm-ts/src/main/kotlin/org/usvm/machine/expr/CallApproximations.kt

```diff
@@ -0,0 +1,879 @@
+    check(expr.args.isEmpty()) {
+        "Array.pop() should have no arguments, but got ${expr.args.size}"
+    }
+
+    scope.calcOnState {
+        // Read the length of the array
+        val lengthLValue = mkArrayLengthLValue(array, arrayType)
+        val length = memory.read(lengthLValue)
+
+        // Decrease the length of the array
+        // TODO: Only decrease the length if it is not zero.
+        //       It is not an error/exception to pop from an empty array!
+        //       If the array is empty, `pop` returns `undefined`.
+        val newLength = mkBvSubExpr(length, 1.toBv())
+
+        // Read the last element of the array (to be removed)
+        val lastIndexLValue = mkArrayIndexLValue(
+            sort = elementSort,
+            ref = array,
+            index = newLength,
+            type = arrayType,
+        )
+        // TODO: If the array is empty, return `undefined` instead of the last element.
+        val removedElement = memory.read(lastIndexLValue)
+
+        // Update the length of the array (AFTER reading the last element)
+        memory.write(lengthLValue, newLength, guard = trueExpr)
+
+        // Return the removed element
+        removedElement
+    }
+}
+
+/**
+ * Handles the `Array.fill(value [,start [,end]])` method call.
+ * Fills the array with the specified `value` from the `start` index to the `end` index.
+ * If `start` index is not provided, it defaults to `0`.
+ * If `end` index is not provided, it defaults to the length of the array.
+ * This method modifies the array in place and returns it.
+ *
+ * https://tc39.es/ecma262/multipage/indexed-collections.html#sec-array.prototype.fill
+ */
+private fun TsExprResolver.handleArrayFill(
+    expr: EtsInstanceCallExpr,
+    arrayType: EtsArrayType,
+    elementSort: USort,
+): UExpr<*>? = with(ctx) {
+    val array = resolve(expr.instance)?.asExpr(addressSort) ?: return null
+    check(expr.args.size == 1) {
+        "Array.fill() should have exactly one argument, but got ${expr.args.size}"
+    }
+    // TODO: support `start` and `end` indices
+    val value = resolve(expr.args.single()) ?: return null
+
+    scope.calcOnState {
+        // Allocate an array to fill
+        val resultArray = memory.allocConcrete(arrayType)
+
+        // Read the length of the original array
+        val lengthLValue = mkArrayLengthLValue(array, arrayType)
+        val length = memory.read(lengthLValue)
+
+        // Fill the result array with the specified `value`
+        memory.initializeArray(
+            resultArray,
+            arrayType,
+            elementSort,
+            sizeSort,
+            (0 until ARRAY_FILL_MAX_SIZE).asSequence().map { value.asExpr(elementSort) }
+        )
+
+        // Set the length of the result array to match the original array
+        val resultLengthLValue = mkArrayLengthLValue(resultArray, arrayType)
+        memory.write(resultLengthLValue, length, guard = trueExpr)
```

Comment:
Why? The resultsArray is a temporary array that doesn't exists anywhere except for this function, you can omit his length information